### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by CDobbins1**

### CODE REVIEW

**Comment by kenny2892** [comment]
I will review this.

File: SecretSanta/test/SecretSanta.Business.Tests/AutomapperProfileConfigurationTests.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            }
+        }
+
+        [TestMethod]
+        public void Map_Gift_SuccessWithNoIdMapped()
+        {
+            User user = new User("TestFirst", "TestLast");
+            User user2 = new User("TestFirst2", "TestLast2");
+            (Gift source, Gift target) = (
+                new MockGift(42, "TestTitle", "TestDesc", "www.Test.com", user), new MockGift(0, "Invalid", "Invalid", "Invalid", user2));
```
Comment (kenny2892) [inline SecretSanta/test/SecretSanta.Business.Tests/AutomapperProfileConfigurationTests.cs]: You could use your SampleData gift as the source.

File: SecretSanta/test/SecretSanta.Business.Tests/AutomapperProfileConfigurationTests.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            Assert.AreEqual<string>(source.Title, target.Title);
+            Assert.AreEqual<string>(source.Description, target.Description);
+            Assert.AreEqual<string>(source.Url, target.Url);
+            Assert.AreEqual<string>(source.User.FirstName, target.User.FirstName);
+        }
+
+        [TestMethod]
+        public void Map_User_SuccessWithNoIdMapped()
+        {
+            User source = new MockUser(42, "TestFirst", "TestLast");
```
Comment (kenny2892) [inline SecretSanta/test/SecretSanta.Business.Tests/AutomapperProfileConfigurationTests.cs]: You could use your SampleData user as the source.

**Comment by MarkMichaelis** [comment]
# Assignment 3

## Main Assignment

- Connect Azure DevOps up for your fork.
- Add a Status Badge (see [Add a status badge to your repository](https://docs.microsoft.com/en-us/azure/devops/pipelines/create-first-pipeline?view=azure-devops&tabs=browser%2Ctfs-2018-2#add-a-status-badge-to-your-repository)) to the `README.md` file in `master` that shows the build status on each branch. ✔
- Be sure that clicking on the badge successfully navigates one to the Azure DevOps pipeline. ✔
- Rebase to the Readme.md update on `master` to the `Assignment3` branch so it too shows the badge (be sure it points to the Assignment3 branch status) ✔
- Appropriately encapsulate Id and all fingerprint data. ✔
- Add non-default constructors to Gift and address all warnings (see BlogEngine's Post.cs file). ✔
- Add any missing entity framework CRUD unit tests for `Gift`. ✔
- Implement an Automapper Configuration between two `Gift` objects. ✔
  - Do not map the Id ✔
- Create services for (do not map primary key Id for all entities):
  - Groups ✔
  - Users ✔
  - Gifts ✔
  - When retrieving the Gift object from the service, User should be pre-populated (included) ✔
- Refactor out SampleData (see BlogEngine's SampleData.cs) ✔

## Extra Credit

- Write exceptional code that is clean, conforms to all coding guidelines, is well refactored, and easy to understand.

## Project

| ProjectName | Nullable | Solutions | Analyzers | Unexpected Disabled Warnings | Waning Level |
| -- | -- | -- | -- | -- | -- |
| SecretSanta.Api | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Business | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Data | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Web | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Api.Tests | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Business.Tests | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Data.Tests | Enable | SecretSanta.sln |  |  | 4 |
| SecretSanta.Web.Tests | Enable | SecretSanta.sln |  |  | 4 |


| Project Analysis |   |
| -----------------| - |
| Build/Compile | (Build succeeded: 0 Error(s), 0 Warning(s)) |
| Disabled Warnings **** | **✔**
| Nullable Enabled: | **✔** |
| Appropriately handle all warnings | **✔** |
| Test Run Succeeded: Total tests=35, Passed=35, Failed=0, Skipped=0 |

**Pull request closed without merging**